expression-src/spec/language/std-functions/DataFunctionsTest.cls

```diff
@@ -20,4 +20,62 @@ private class DataFunctionsTest {
         Object result = Evaluator.run(formula);
         Assert.isNotNull(result);
     }
-}
+    
+    @IsTest
+    private static void parseJsonWithAccount() {
+        Account acc = new Account(
+            Name = 'Test Account',
+            Description = '{"name":"John","age":30}'
+        );
+        insert acc;
+        
+        acc = [SELECT Id, Description FROM Account WHERE Id = :acc.Id LIMIT 1];
```

Comment:
There's no need to re-query the record here. Salesforce automatically updates the reference's `Id` property when the `insert` operation happens, so the Id will be present without this.